### PR TITLE
Integrate CodeQL to CI

### LINKED ISSUE #812: Integrate CodeQL to CI
### Description

Update existing codeql integration.

### Acceptance Criteria
 - Codeql should analyze for typescript
 - Codeql invocation should avoided for unrelated directory changes
 

Please describe the conditions which must be met for this issue to close

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?

This PR resolves #812  

### How was it solved?

- [x] Update codeql coverage in github action

### How was it tested?
- [x] ci

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        language: ['javascript']
+        language: ['typescript']
```
Comment (sameersubudhi) [inline .github/workflows/codeql.yml:35]:
```suggestion
        language: ['javascript']
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (priojeetpriyom) [reply .github/workflows/codeql.yml:35]: Discussed offline.

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ on:
     branches: ['development', feature/*, hotfix/*, main, release/*]
   pull_request:
     # The branches below must be a subset of the branches above
-    branches: ['development', feature/*, hotfix/*, release/*]
+    branches: ['development', feature/*, hotfix/*, main, release/*]
```
Comment (sameersubudhi) [inline .github/workflows/codeql.yml:19]:
```suggestion
    branches: ['development', feature/*, hotfix/*, release/*]
```

**Review by sameersubudhi — APPROVED** [review]

**Review by nagdahimanshu — APPROVED** [review]
LGTM 👍

**Pull request merged**